frontend/javascripts/oxalis/model/sagas/mapping_saga.js

```diff
@@ -24,18 +25,55 @@ import {
   getAgglomeratesForDatasetLayer,
 } from "admin/admin_rest_api";
 import type { APIMapping } from "types/api_flow_types";
-import { getLayerByName } from "oxalis/model/accessors/dataset_accessor";
+import { getLayerByName, getMappingInfo } from "oxalis/model/accessors/dataset_accessor";
 import { type Mapping } from "oxalis/store";
 import ErrorHandling from "libs/error_handling";
 import { MAPPING_MESSAGE_KEY } from "oxalis/model/bucket_data_handling/mappings";
+import api from "oxalis/api/internal_api";
+import { MappingStatusEnum } from "oxalis/constants";
 
 type APIMappings = { [string]: APIMapping };
 
+const isAgglomerate = mapping => {
+  if (!mapping) {
+    return false;
+  }
+  return mapping.mappingType === "HDF5";
+};
+
 export default function* watchActivatedMappings(): Saga<void> {
+  const oldActiveMappingByLayer = {};
```

Comment:
Agreed, but since we're using action channels I'd argue that the initialization should stay in line 45 since immediately after that the action channels start picking up actions.

Reply:
Ok, sounds good :+1: